src/update_roster.R

```diff
@@ -1,642 +0,0 @@
-          fill_ids()
-
-        roster <- roster |>
-          dplyr::select(-c(height)) |>
-          dplyr::left_join(
-            shield |>
-              dplyr::select(gsis_id, birth_date, height, headshot_gsis = headshot) |>
-              dplyr::distinct(),
-            by = c("gsis_id"),
-            na_matches = "never"
-          ) |>
-          dplyr::mutate(headshot = dplyr::coalesce(headshot, headshot_gsis)) |>
-          dplyr::select(-c(headshot_gsis)) |>
-          fill_ids()
-      }
-    }
-
-    if (season >= 2016 && nrow(weekly_rosters) == 0) {
-      roster <- shield
-    }
-
-    if ((season < 2002 && length(roster) == 0) |
-        (season >= 2002 && length(weekly_rosters) == 0)) {
-      return(invisible(NULL))
-    }
-
-    if (season >= 2002) {
-      weekly_rosters <- weekly_rosters |>
-        dplyr::mutate(dplyr::across(c(gsis_it_id, smart_id), as.character),
-                      dplyr::across(c(entry_year, rookie_year), as.integer)) |>
-        nflreadr:::join_coalesce(df_players,
-                                 by = c("gsis_id"),
-                                 na_matches = "never")
-      # fix for missing gsis_ids
-      weekly_rosters_missing <- weekly_rosters |>
-        dplyr::filter(is.na(gsis_id))
-      if (nrow(weekly_rosters_missing)) {
-        weekly_rosters_missing <- weekly_rosters_missing |>
-          tibble::as_tibble() |>
-          nflreadr:::join_coalesce(df_players,
-                                   by = c("gsis_it_id"),
-                                   na_matches = "never")
-        weekly_rosters <- weekly_rosters |>
-          dplyr::filter(!is.na(gsis_id)) |>
-          dplyr::bind_rows(weekly_rosters_missing)
-      }
-      weekly_rosters <- weekly_rosters |>
-        dplyr::mutate(years_exp = as.integer(season) - as.integer(entry_year)) |>
-        dplyr::select(dplyr::any_of(
-          c(
-            "season",
-            "team",
-            "position",
-            "depth_chart_position",
-            "jersey_number",
-            "status",
-            "full_name",
-            "first_name",
-            "last_name",
-            "birth_date",
-            "height",
-            "weight",
-            "college",
-            "gsis_id",
-            "espn_id",
-            "sportradar_id",
-            "yahoo_id",
-            "rotowire_id",
-            "pff_id",
-            "pfr_id",
-            "fantasy_data_id",
-            "sleeper_id",
-            "years_exp",
-            "headshot_url",
-            "ngs_position",
-            "week",
-            "game_type",
-            "status_description_abbr",
-            "football_name",
-            "esb_id",
-            "gsis_it_id",
-            "smart_id",
-            "entry_year",
-            "rookie_year",
-            "draft_club",
-            "draft_number"
-          )
-        ))
-      cli::cli_alert_info("Save weekly rosters...")
-      nflversedata::nflverse_save(
```

Comment:
I also moved the save logic outside of the scope of the build rosters function and into the exec script so that it would be easier to debug outputs later on without uploading each time

Reply:
yeah that's good. We should separate data creation from uploads completely